build-scripts/collect_remediations.py

```diff
@@ -117,6 +120,16 @@ def main():
     product = ssg.utils.required_key(env_yaml, "product")
     output_dirs = prepare_output_dirs(args.output_dir, args.remediation_type)
 
+    for platform_file in os.listdir(args.platforms_dir):
+        platform_path = os.path.join(args.platforms_dir, platform_file)
+        try:
+            platform = ssg.build_yaml.Platform.from_yaml(platform_path, env_yaml)
+        except ssg.build_yaml.DocumentationNotComplete:
+            # Happens on non-debug build when a platform is
+            # "documentation-incomplete"
+            continue
+        env_yaml["product_cpes"].platforms[platform.name] = platform
```

Comment:
I don't like that we add it to the env_yaml. It's a dictionary already containing multiple different types of data from various different sources. There is arbitrary data with no structure. We use it as universal storage. Adding CPE to it increases already large confusion. Instead, we should define the structure of the data and keep everything organized. Try separating it from env_yaml, at least on high level.